### ISSUE TITLE
`import-x/no-duplicates` false positive with `declare module`

### ISSUE THREAD

**Issue by DavidArchibald** [un-ts/eslint-plugin-import-x#367]
`import-x/no-duplicates` has a false positive when two imports resolve to the same file but contains `declare module`. For instance, these imports trigger `no-duplicates`:
```ts
import { createEventDispatcher } from "svelte";
import { flip } from "svelte/animate";
```

However the file has the structure:
```ts
declare module 'svelte' {
    ...
}

declare module 'svelte/animate' {
    ...
}
```

Which means `svelte` and `svelte/animate` are not interchangeable.

**Comment by JounQin** [comment]
Please search before rasing new issues.
